src/types/ErrorRate/__tests__/utils.test.ts

```diff
@@ -0,0 +1,77 @@
+import * as U from '../';
+import { ToleranceConfig } from '../../ServerConfig';
+
+const defaultTolerance = (): ToleranceConfig => {
+  return {
+    code: new RegExp(''),
+    degraded: U.DEFAULTCONF.degraded,
+    failure: U.DEFAULTCONF.failure,
+    protocol: U.DEFAULTCONF.protocol,
+    direction: U.DEFAULTCONF.direction
+  };
+};
+
+describe('Utils library', () => {
+  describe('Parse Annotation', () => {
+    it('should return tolerances with defaults if only code', () => {
+      var tol = defaultTolerance();
+      tol.code = new RegExp('[45]');
+      expect(U.parseAnnotationTEST('[45]')).toEqual(tol);
+    });
+
+    it('should replace x or X by d', () => {
+      var tol = defaultTolerance();
+      tol.code = new RegExp('[45]d[2]');
+      expect(U.parseAnnotationTEST('[45]x[2]')).toEqual(tol);
+      expect(U.parseAnnotationTEST('[45]X[2]')).toEqual(tol);
```

Comment:
duplicate line ? or would you expect add some test, as commented, I thik that the replace of x by d is broken.
An easy way to validate it could be to add a test of a matching with the resulting parsed annotation.

(That would help any other maintainers to understand it; nothing wrong, just thinking loud minor changes that may help in that sense).